### PR TITLE
Subentity detail

### PR DESCRIPTION
**Pull request by orangejenny**
Roundabout fix for http://manage.dimagi.com/default.asp?167204

Support details that specify a nodeset. This nodeset gets expanded in the context of the case, and the detail displays a row for each node in the expanded set. Useful for mobile UCR, where the main detail object is the report and we then want to display detail with a line for each row in the report.

### CODE REVIEW

**Comment by orangejenny** [comment]
dependent on https://github.com/dimagi/commcare/pull/150

File: app/src/org/commcare/android/adapters/EntitySubnodeListAdapter.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+import java.util.List;
+
+/**
+ * Created by jschweers on 8/24/2015.
+ * <p/>
+ * Adapter for taking a nodeset, contextualizing it against an entity,
+ * and then displaying one item for each node in the resulting set.
+ */
+public class EntitySubnodeListAdapter implements ListAdapter {
```
Comment (orangejenny) [inline app/src/org/commcare/android/adapters/EntitySubnodeListAdapter.java]: Better naming suggestions would be welcome.

File: app/src/org/commcare/android/adapters/EntitySubnodeListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        EntityView view = (EntityView) convertView;
+        Entity entity = this.entities.get(position);
+        if (view == null) {
+            view = new EntityView(context, detail, entity, null, null, position, false);
+        } else {
+            view.refreshViewsForNewEntity(entity, false, position);
+        }
+        int color = this.rowColors[position % 2];
+        if (color != -1) {
+            view.setBackgroundColor(color);
```
Comment (orangejenny) [inline app/src/org/commcare/android/adapters/EntitySubnodeListAdapter.java]: The striping in the regular entity detail activity is several times more complicated than this...not sure why. Tested that this works fine as you scroll and views get reused.

**Comment by orangejenny** [comment]
Updated screenshot:
screenshot_2015-08-26-10-31-18

File: app/res/layout/entity_detail_list.xml [hunk 3]
```diff
@@ -4,9 +4,18 @@
               android:layout_height="match_parent"
               android:orientation="vertical"
     >
+
+    <LinearLayout
+        android:id="@+id/entity_detail_header"
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:layout_above="@+id/screen_entity_detail_list"
+        android:orientation="horizontal">
```
Comment (ctsims) [inline app/res/layout/entity_detail_list.xml]: The default visibility here should be "gone", right?

**Comment by ctsims** [comment]
This looks pretty good to me. One question: The EntityDetailFragment is already fairly limited in scope, and it seems really confusing to wire it together in the same class. Is it possible to separate the concerns, such that there's one fragment for contextualized entity views, and one for flat entity views?

File: app/res/layout/entity_detail_list.xml (hunk 3 above)
Comment (orangejenny) [inline app/res/layout/entity_detail_list.xml]: Visibility gets set in the code (whether VISIBLE or GONE), doe sit matter if there's a default value here?

Comment (ctsims) [inline app/res/layout/entity_detail_list.xml]: It matters a bit due to the way the stuff gets displayed in views, but possibly not enough that we should worry about it

Comment (orangejenny) [inline app/res/layout/entity_detail_list.xml]: will fix it, just hadn't known it made a difference

**Comment by orangejenny** [comment]
@ctsims good idea. carved out an EntitySubnodeDetailFragment subclass.

File: app/src/org/commcare/android/view/EntityView.java [hunk 4]
```diff
@@ -86,35 +87,42 @@ public EntityView(Context context, Detail d, Entity e, TextToSpeech tts,
...
         super(context);
         this.context = context;
         this.views = new View[headerText.length];
         this.mHints = d.getHeaderSizeHints();
         String[] headerForms = d.getHeaderForms();
-        
-        for (int i = 0 ; i < views.length ; ++i) {
+
+        int[] colors = AndroidUtil.getThemeColorIDs(context, new int[]{R.attr.entity_view_header_background_color, R.attr.entity_view_header_text_color});
+        Log.i("DEBUG-i", "Background color is: " + colors[0] + ", text color is: " + colors[1]);
```
Comment (ctsims) [inline app/src/org/commcare/android/view/EntityView.java]: hanging debug

**Pull request merged**